src/holofuel/pages/CreateRequest/CreateRequest.test.js

```diff
@@ -227,6 +229,57 @@ describe('CreateRequest', () => {
     expect(within(getByTestId('counterparty-nickname')).getByText('Loading')).toBeInTheDocument()
   })
 
+  it('renders a clickable list of recent counterparties', async () => {
+    const agent1 = {
+      id: 'fkljd',
+      nickname: 'Jo'
+    }
+
+    const agent2 = {
+      id: 'dskajln',
+      nickname: 'Bob'
+    }
+
+    const mocks = [
+      {
+        request: {
+          query: HolofuelCounterpartyQuery,
+          variables: { agentId: agent1.id }
+        },
+        result: {
+          data: { holofuelCounterparty: agent1 }
+        }
+      },
+      {
+        request: {
+          query: HolofuelHistoryCounterpartiesQuery
+        },
+        result: {
+          data: { holofuelHistoryCounterparties: [agent1, agent2] }
+        }
+      }
+    ]
+
+    const { getAllByTestId, getByLabelText } = await renderAndWait(<MockedProvider mocks={mocks} addTypename={false}>
+      <CreateRequest history={{}} />
+    </MockedProvider>)
+
+    const agentRows = getAllByTestId('agent-row')
+
+    const agent1Row = agentRows.find(agentRow => {
+      const { queryByText } = within(agentRow)
+      return !!queryByText(agent1.nickname)
+    })
```

Comment:
I don't love this way of getting at agent1Row, but as far as I can tell, it's the best way of doing this.

Reply:
Since we know we'll be testing the first agent, we can also expect (and test that) the first row in agentRows is the parent row of agent1 nickname.  If we go with this approach we could also do something like the following to get the full agent1Row:

```
   const { getAllByTestId, getByLabelText } = await renderAndWait(<MockedProvider mocks={mocks} addTypename={false}>
      <CreateOffer history={{}} />
    </MockedProvider>)

    const agentRows = getAllByTestId('agent-row')
    const agent1Row = agentRows[0]
    const { queryByText, getByText } = within(agent1Row)

    expect(queryByText(agent1.nickname)).toBeInTheDocument()
    expect(agent1Row).toBeInTheDocument()

    fireEvent.click(agent1Row)

    expect(getByLabelText('To').value).toEqual(agent1.id)
    expect(getByText('Selected')).toBeInTheDocument()
```

I'm fine with either approach though.

Reply:
right. The two test slightly different things. Your test is slightly more determined than mine, in that yours tests that agent1 appears in the first row, where mine just tests that it appears in one of the rows. I agree it doesn't really matter for this test.